tools/rtpsrelay/Relay.idl

```diff
@@ -5,21 +5,85 @@
  * See: http://www.opendds.org/license.html
  */
 
+#include "dds/DdsDcpsCore.idl"
+
 module RtpsRelay {
 
+  // Vendor Id value from RTPS specification.
+  typedef octet GuidVendorId_t[2];
+
+  // GUID prefix bytes - uniquely specify the participant.
+  typedef octet GuidPrefix_t[12];
+
+  // Key portion of Entity identifier.
+  typedef octet EntityKey_t[3];
+
+  // Uniquely identify each entity within a participant.
+  struct EntityId_t {
+    @key EntityKey_t entityKey;
+    @key octet entityKind;
+  };
+
+  const octet ENTITYKIND_BUILTIN_UNKNOWN         = 0xc0;
+  const octet ENTITYKIND_BUILTIN_PARTICIPANT     = 0xc1;
+  const octet ENTITYKIND_BUILTIN_WRITER_WITH_KEY = 0xc2;
+  const octet ENTITYKIND_BUILTIN_WRITER_NO_KEY   = 0xc3;
+  const octet ENTITYKIND_BUILTIN_READER_NO_KEY   = 0xc4;
+  const octet ENTITYKIND_BUILTIN_READER_WITH_KEY = 0xc7;
+  const octet ENTITYKIND_BUILTIN_TOPIC           = 0xc5;
+
+  const octet ENTITYKIND_USER_UNKNOWN         = 0x00;
+  const octet ENTITYKIND_USER_WRITER_WITH_KEY = 0x02;
+  const octet ENTITYKIND_USER_WRITER_NO_KEY   = 0x03;
+  const octet ENTITYKIND_USER_READER_NO_KEY   = 0x04;
+  const octet ENTITYKIND_USER_READER_WITH_KEY = 0x07;
+
+  // Vendor specific Kinds have the following bit pattern: 01xx_xxxx
+  const octet ENTITYKIND_OPENDDS_SUBSCRIBER   = 0x41;
```

Comment:
If the OpenDDS-specific ones aren't needed I'd rather not have them here.  In general I don't know that we need any `const` declarations here, they're already in libOpenDDS_Rtps.